low-level/files/opt/seagate/sspl/bin/feed_sspl_conf_to_consul.py

```diff
@@ -0,0 +1,240 @@
+                'password': mc_passwd or self._get_result(
+                        'consul kv get storage_enclosure/controller/password'),
+                'primary_mc': {
+                    'ip':  primary_mc or self._get_result(
+                        'consul kv get storage_enclosure/controller/primary_mc/ip'),
+                    'port': primary_mc_port or self._get_result(
+                        'consul kv get storage_enclosure/controller/primary_mc/port')
+                },
+                'secondary_mc': {
+                    'ip': secondary_mc or self._get_result(
+                        'consul kv get storage_enclosure/controller/secondary_mc/ip'),
+                    'port': secondary_mc_port or self._get_result(
+                        'consul kv get storage_enclosure/controller/secondary_mc/port')
+                }
+            },
+            'type': storage_type or self._get_result(
+                'consul kv get storage_enclosure/type'),
+        }
+        # BMC information
+        parser['BMC'] = { node: {
+                            'ip': bmc_ip or self._get_result(f'consul kv get bmc/{node}/ip'),
+                            'user': bmc_user or self._get_result(f'consul kv get bmc/{node}/user'),
+                            'secret': bmc_passwd or self._get_result(f'consul kv get bmc/{node}/secret')
+                            }
+                        }
+        # Update same password for required sections
+        if rmq_passwd:
+            self.rmq_encrypted_passwd = rmq_passwd
+        for section in self.rmq_same_pass_sect:
+            parser[section] = {"username": rmq_user}
+            parser[section] = {"password": self.rmq_encrypted_passwd}
+        # Insert config data
+        prefix = component.lower() + '/config/'
+        for k, v in parser.items():
+            if k.lower() in self.common_config_sect:
+                self._insert_nested_dict_to_consul(v, prefix=k.lower())
+            k = prefix + k
+            self._insert_nested_dict_to_consul(v, prefix=k)
+        # Update persistent data file
+        persistent_data = f"cluster_id={self.cluster_id}\n" + \
+                          f"node_id={self.node_id}\n" + \
+                          f"rmq_encrypted_passwd={self.rmq_encrypted_passwd}"
+        with open(self.persistent_data_file, "w") as fObj:
+            fObj.write(persistent_data)
+        print(persistent_data)
+
+
+if __name__ == '__main__':
+    argParser = argparse.ArgumentParser(
+        usage = "%(prog)s [-h] [-F] [-C] [-N] [-Ru] [-Rp] [-A] [-B] \
+                    [-Ap] [-Bp] [-U] [-P] [--bmc_ip] [--bmc_user] [--bmc_passwd]",
+        formatter_class = argparse.RawDescriptionHelpFormatter)
+    argParser.add_argument("-F", "--config_file", default="/etc/sspl.conf", help="Config file path")
+    argParser.add_argument("-C", "--component", default="sspl", help="Keys will be stored with this prefix")
```

Comment:
what other component possible ?

Reply:
Added this as per my initial plan if we maintain common config and component config are separately. 
New commit carries the changes.